meson.build

```diff
@@ -55,6 +55,14 @@ add_project_arguments('-Wno-reorder-init-list', language : ['cpp'])
 add_project_arguments('-Wno-unknown-warning-option', language : ['cpp'])
 add_project_arguments('-Wno-unused-parameter', language : ['c', 'cpp'])
 
+if get_option('buildtype') == 'release'
+  add_project_arguments('-O3',  language : ['c', 'cpp'])
```

Comment:
If O3 is incompatible with hardening, shouldn't this be in the `else` clause of checking hardening is enabled (which is by default)?